D3D12HelloWorld/Graphics/PipelineState.cs

```diff
@@ -12,24 +12,30 @@ public sealed class PipelineState : IDisposable {
 
         internal ID3D12PipelineState NativePipelineState { get; }
 
-        public PipelineState(GraphicsDevice device, ID3D12RootSignature rootSignature, byte[] computeShader)
-            : this(device.NativeDevice, rootSignature, CreateComputePipelineStateDescription(rootSignature, computeShader)) {
+        public PipelineState(GraphicsDevice device, ID3D12RootSignature rootSignature, byte[] computeShader, string? name = null)
+            : this(device.NativeDevice, CreateComputePipelineStateDescription(rootSignature, computeShader), name) {
         }
 
-        internal PipelineState(ID3D12Device device, ID3D12RootSignature rootSignature, ComputePipelineStateDescription pipelineStateDescription) {
+        internal PipelineState(ID3D12Device device, ComputePipelineStateDescription pipelineStateDescription, string? name) {
             IsCompute = true;
-            RootSignature = rootSignature;
+            RootSignature = pipelineStateDescription.RootSignature ?? throw new ArgumentException($"{nameof(ComputePipelineStateDescription.RootSignature)} may not be null.", nameof(pipelineStateDescription));
             NativePipelineState = device.CreateComputePipelineState(pipelineStateDescription);
+            if (!string.IsNullOrWhiteSpace(name)) {
+                NativePipelineState.Name = name;
+            }
```

Comment:
A minor convenience, as I'm adding all over the place to provide names to facilitate debugging.